OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -131,41 +136,98 @@ public override Activity Tick(Actor self)
 			if (IsCanceled || self.IsDead)
 				return NextActivity;
 
-			if (!isCalculated)
+			if (aircraft.Info.VTOL)
+				TryToReserveResupplier(self);
+			else if (!isCalculated)
 				Calculate(self);
 
 			if (dest == null || dest.IsDead)
```

Comment:
HeliRTB also checked `Reservable.IsReserved(dest)`. Is the omission here intentional?

Reply:
Actually no, this is no different from HeliRTB.

If you look at HeliRTB again, what you're referring to is basically what `TryToReserveResupplier` (and `Calculate` by calling it) above does now, and that one indeed checks whether the old dest is reserved, while only returning unreserved actors.

If `dest` isn't null or dead here, it means either `ChooseResupplier(self, true)` was successful in returning a valid, unreserved resupplier, or `dest` is carried over from a previous tick, in which case the only actor that can have reserved it should be this one.